### DISCUSSION TITLE
Redirect all pages to a particular page if user logged out(missing in session)?

### DISCUSSION THREAD

**Discussion by viraj071** [nextauthjs/next-auth#4976] [Help] [answered]
I am using next-auth.js in my next.js project.
I want to redirect all pages to the "/" page or login page if the user logs out on one of the open pages of the app. How can I do this?

**Comment by balazsorban44** [comment, marked as answer]
Middleware lets you do exactly that! Have a look: https://next-auth.js.org/configuration/nextjs#basic-usage

**Reply by viraj071** [reply]
I am already using the middleware to redirect to the login page on new requests. How do I make it work on the open pages?

**Reply by balazsorban44** [reply]
This should work, I believe:

```js
const { status } = useSession()
const router = useRouter()

useEffect(() => {
  status === "unauthenticated" && router.push("/")
}, [status, router])
```

**Reply by viraj071** [reply]
Yes. The above should work but I will need to add the above to every component under protected paths. Wondering if there is a centralized place I can configure this.

**Reply by balazsorban44** [reply]
It was late yesterday, I forgot we had this 😅:

https://next-auth.js.org/getting-started/client#specifying-a-callbackurl-1


So you can just call `signOut({callbackUrl: "/"})`

**Reply by viraj071** [reply]
Yes. I am already doing this. This would redirect the page on which we click signOut. The code you mentioned earlier will have to go in every protected page right?

**Reply by balazsorban44** [reply]
The useEffect can be put higher up in the tree like in `_app`, just make sure it's still under `SessionProvider` to make `useSession` work.

> This would redirect the page on which we click signOut. 



Both of these aren't a problem if you have a shared layout like a navbar for example.
